src/v/cluster/rm_stm.cc

```diff
@@ -259,6 +285,26 @@ struct tx_snapshot_v2 {
     fragmented_vector<rm_stm::seq_entry> seqs;
 };
 
+struct tx_snapshot_v3 {
```

Comment:
im concerned about test coverage. i don't see this type referenced in any unit test files. am i missing it? seems like there needs to be one or more data compatibility tests.

Reply:
This struct hasn't changed, it was moved from rm_stm.h, if you compare it with `rm_stm::tx_snapshot` before the change - it's exactly the same so we don't need a test specifically for `tx_snapshot_v3` but you're right that compatibility are needed.

@ZeDRoman can you please add two tests:

positive test:
  - leadership balancing and partition migration is off
  - redpanda cluster has mixed versions (head and latest release)
  - move leader of a data partition to a node with a newest version
  - execute several transacitons
  - check that the leadership hasn't changed
  - downgrade the node
  - check that it's possible to read the written txes

negative test:
  - redpanda cluster has the same head version
  - execute several transacitons
  - downgrade the cluster
  - if it doesn't fail on start - try to read the written txes with an expectation to fail